packages/babel-plugin-transform-modules-commonjs/src/index.js

```diff
@@ -132,14 +143,26 @@ export default function(api, options) {
 
             let header;
             if (isSideEffectImport(metadata)) {
+              if (metadata.lazy) throw new Error("Assertion failure");
```

Comment:
I guess this should never really happen, but maybe we could make the error a little bit more descriptive?

Reply:
For assertions I don't usually bother. I'd just `assert(!metadata.lazy)` but usually that isn't enough to make Flowtype happy so I've kind of got into the habit of doing it like this. I really just threw this in for us.